rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+                """Get vendors for the given pids."""
+                fields = ['pid', 'name']
+                return VendorsSearch() \
+                    .get_records_by_terms(terms=vendor_pids, fields=fields,
+                                          as_dict=True)
+
+            headers = dict.fromkeys(self.csv_included_fields)
+
+            # write the CSV output in memory
+            line = Line()
+            writer = csv.DictWriter(line,
+                                    quoting=csv.QUOTE_ALL,
+                                    fieldnames=headers)
+            writer.writeheader()
+            yield line.read()
+
+            for pids, order_batch_results in batch(search_result):
+                order_lines, documents, accounts, receipt_lines = \
+                    get_linked_records_by_order_pids(pids)
+                receipts = get_receipts_by_order_pids(pids)
+                # vendors
+                vendor_pids = [order['vendor']['pid']
+                               for order in order_batch_results.values()
+                               if order['vendor']['pid'] not in vendors]
+                vendors.update(get_vendors_by_pids(vendor_pids))
+
+                # prepare export based on order lines
+                for order_line_pid, order_line in order_lines.items():
+                    order_pid = order_line['acq_order']['pid']
+                    order_data = order_batch_results[order_pid]
+                    vendor_data = vendors.get(order_data['vendor']['pid'])
+
+                    csv_data = {
+                        k: order_data.get(f) for k, f in order_fields.items()
+                    }
+
+                    # Update csv data with vendor
+                    csv_data['vendor_name'] = vendor_data.get('name')
+
+                    # extract order notes
+                    order_notes = filter(
+                        lambda x: x.get('source').get('type') == 'acor',
+                        order_data.get('notes', {})
+                    )
+                    for note in order_notes:
+                        note_type = note.get('type')
+                        column_name = f'order_{note_type}'
+                        csv_data[column_name] = note.get('content')
+
+                    # update csv data with document infos
+                    csv_data.update(
+                        documents.get(order_line['document']['pid']))
+
+                    # update csv data with account infos
+                    account = accounts.get(order_line['acq_account']['pid'])
+                    csv_data.update({
+                        'account_name': account.get('name'),
+                        'account_number': account.get('number'),
+                    })
+
+                    # update csv data with order line infos
+                    csv_data.update({
+                        'order_lines_priority': order_line.get('priority'),
+                        'order_lines_notes': ' | '.join(
```

Comment:
😝 this will be very unreadable but do we have any better solution.... i don't think so

Reply:
🤓